toolchain/mfc/viz/reader.py

```
@@ -0,0 +1,387 @@
+    """Detect endianness from the first record marker (should be 16 for header)."""
+    with open(path, 'rb') as f:
+        raw = f.read(4)
+    le = struct.unpack('<i', raw)[0]
+    if le == 16:
+        return '<'
+    be = struct.unpack('>i', raw)[0]
+    if be == 16:
+        return '>'
+    raise ValueError(
+        f"Cannot detect endianness: first record marker is {le} (LE) / {be} (BE), expected 16"
+    )
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)